src/Tools/ExternalAccess/FSharp/Navigation/IFSharpDocumentNavigationService.cs

```diff
@@ -4,42 +4,38 @@
 
 #nullable disable
 
+using System.Threading;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Navigation;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Navigation
 {
     internal interface IFSharpDocumentNavigationService : IWorkspaceService
     {
-        /// <summary>
-        /// Determines whether it is possible to navigate to the given position in the specified document.
-        /// </summary>
         bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan);
+        /// <inheritdoc cref="IDocumentNavigationService.CanNavigateToSpan"/>
+        bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, CancellationToken cancellationToken);
```

Comment:
@cartermp Adding new overloads that take cancellation tokens.  Existing overloads will remain for binary compat.  Once you can update your code over to the new overloads, we can remove the old ones.